lib/plugins/threaded-signing/plugin.c

```diff
@@ -111,16 +118,26 @@ local_signature_info_create(const signature_info_t *signature_info)
 }
 
 /* Frees all allocated memory and resets members. Excluded are the worker thread members |thread|,
- * |mutex|, |cond| and |is_running|, but also the counter |nbr_of_unsigned_hashes|. */
+ * |mutex|, |cond| and |is_running|. */
 static void
 sv_threaded_plugin_reset(sv_threaded_plugin_t *self)
 {
   local_signature_info_free(self->signature_info);
   self->signature_info = NULL;
-  free(self->hash_to_sign);
-  self->hash_to_sign = NULL;
+
+  const int output_buffer_end = self->output_buffer_idx;
+  for (int i = 0; i < output_buffer_end; i++) {
+    free(self->output_buffer[i].signature);
+    self->output_buffer[i].signature = NULL;
+    self->output_buffer[i].size = 0;
+  }
+
+  const int input_buffer_end = self->input_buffer_idx;
```

Comment:
- How about to remove line 135.
- Use "self->input_buffer_idx" on row 136 instead of "input_buffer_end".
- And then finally add "self->input_buffer_idx = 0" after the for loop.

Please check if this suggestion works.

Reply:
See comment above.